general/glossary.md

```diff
@@ -0,0 +1,60 @@
+# Glossary
+
+This documents aims to provide a common lexicon for terms used to refer to artifacts.
+
+### Motivation
+
+Our technology space is filled with overloaded words like _package_, _module_, and _bundle_.
+
+Despite the potential for confusion, we can make our documentation, APIs, and all forms of written and spoken communication clearer by carefully selecting and qualifying the terminology that we use.
+
+If this document succeeds, we should end up having a documentation where each object/artifact has one single term to designate it. In other words, [Wittgenstein](https://en.wikipedia.org/wiki/Ludwig_Wittgenstein) should be proud of us (see [Tractatus Logico-Philosophicus](https://en.wikipedia.org/wiki/Tractatus_Logico-Philosophicus)).
+
+Note that we also include an [appendix section](#appendix-trademarks--product-names) listing the canonical spelling and casing for used trademarks or product names.
+
+## Technical Terms
+
+1. Frontend realm:
+    1. **JavaScript module**: a generic JavaScript module in any format (AMD, CommonJS, ...).
+        1. **AMD module**: a JavaScript AMD module (the `.js` file or its logical representation at runtime).
+        2. **CommonJS module**: a JavaScript CommonJS module (the `.js` files used in node).
+        3. **ES module**: a module written in ECMAScript 6+ format (i.e.: `import from` syntax).
+        4. ~~ESModule~~: use _ES module_ instead.
+        5. ~~ECMAScript module~~: use _ES module_ instead.
+        6. ~~Harmony module~~: use _ES module_ instead.
+        7. ~~JavaScript module~~: use _ES module_ instead.
+    2. **npm package**: a JavaScript package (those described by a `package.json` file).
+    3. **webpack bundle**: a `.js` file created by `webpack`, containing several bundled `.js` files.
+2. Backend realm:
+    1. **Java package**: a Java package as defined by the language.
```

Comment:
As a non-Java developer, a link here would be helpful. My JS-developer guess is [_I'm not going to write this down because I may look foolish_].